examples/pytorch_to_stockroom_loader.py

```diff
@@ -0,0 +1,161 @@
+    'classification': list(('MNIST FashionMNIST KMNIST EMNIST QMNIST' +
+                            'FakeData LSUM ImageFolder DatasetFolder' +
+                            'ImageNet CIFAR10 CIFAR100 STL10 SVHN' +
+                            'USPS').split(' ')),
+    'captioning': list('CocoCaptions Flickr8k Flickr30k'.split(' ')),
+    #'detection': list('')
+    'segmentation': list('VOCSegmentation SBDataset'.split(' ')),
+    'action_reco': list('UCF101 HMDB51 Kinetics400'.split(' ')),
+}
+
+VISION_COLUMN_NAMES = {
+    'classification': ('image', 'label'),
+    'captioning': ('image', 'caption'),
+    #'detection': None,
+    'segmentation': ('image', 'segmentedImage'),
+    'action_reco': ('video', 'audio', 'label')
+}
+
+def hangar_transform(item):
+    if isinstance(item, torch.Tensor):
+        return item.detach().numpy()
+    if isinstance(item, int):
+        return np.array([item])
+    if isinstance(item, list):
+        subsample = dict()
+        for idx, i in enumerate(item):
+            subsample[idx] = i
+        return subsample
+
+
+def add_from_dataset(
+        name: str,
+        dataset: Dataset,
+        root_dir: Union[Path, str],
+        column_postfixes: list = None
+        ) -> str:
+
+    repo = Repository('./')
+    if not repo.initialized:
+        repo.init(user_name='jj', user_email='jj@g.com', remove_old=True)
+
+    master = repo.checkout(write=True)
+
+    # determine the columns to create
+    sample = dataset[0]
+    columns_added = []
+    for i, column in enumerate(sample):
+
+        if isinstance(column, int):
+            prototype_int = np.array([column])
+            column_name = f'{name}_{column_postfixes[i]}'
+            columns_added.append(column_name)
+            master.add_ndarray_column(
+                column_name,
+                prototype=prototype_int)
+            print(f'Created {column_name}')
+
+        elif isinstance(column, torch.Tensor):
+            numpy_col = column.detach().numpy()
+            column_name = f'{name}_{column_postfixes[i]}'
+            columns_added.append(column_name)
+            master.add_ndarray_column(
+                column_name,
+                prototype=numpy_col)
+            print(f'Created {column_name}')
+
+        elif isinstance(column, list):
+            # nested columns
+            if isinstance(column[0], str):
+                column_name = f'{name}_{column_postfixes[i]}'
+                columns_added.append(column_name)
+                master.add_str_column(column_name,
+                                      contains_subsamples=True)
+                print(f'Created subsampled column {column_name}')
+
+    # commit and close hangar repo
+    master.commit('added columns'+str(columns_added))
+    master.close()
+
+    assert len(sample) == len(columns_added), f'{len(sample)} != {len(columns_added)}'
+    # open stockroom
+    stock = StockRoom('./', write=True)
+
+    with stock:
+        with stock.run():
```

Comment:
why two context managers?

Reply:
outer one for exiting from the checkout 
and inner one for committing all the changes.

Reply:
That's odd. We shouldn't do that. I'll remove the object level context managers from the source (it's a leftover from the previous version) and document the usage in `.run()`. Sorry about the confusion

Reply:
ohh I see, I'll remove it